lib/iotagent-mqtt.js

```diff
@@ -254,11 +254,14 @@ function notificationHandler(device, updates, callback) {
  * @return {Object}                 Parsed message or null if an error has occurred.
  */
 function parseMessage(message) {
-    var parsedMessage;
+    var parsedMessage,
+        stringMessage;
 
     try {
-        parsedMessage = JSON.parse(message.toString());
+        stringMessage = message.toString();
+        parsedMessage = JSON.parse(stringMessage);
     } catch (e) {
+        logger.debug('Parse error treating message: %s', stringMessage);
```

Comment:
Maybe you would like to include the error in the logs somehow to facilitate the inspection of the problem... :)